src/world.js

```diff
@@ -123,6 +123,11 @@ class World extends EventEmitter {
 
   unloadColumn (chunkX, chunkZ) {
     const key = columnKeyXZ(chunkX, chunkZ)
+    if (this.savingQueue.has(key)) {
+      this.savingQueue.delete(key)
+      this.finishedSaving = Promise.all([this.finishedSaving,
+        this.storageProvider.save(chunkX, chunkZ, this.columns[key])])
+    }
```

Comment:
This is wrong, if the chunk was queued for saving it means it needs to be saved, you cannot just remove it from the queue. All you need to do is call `await this.waitSaving()`